src/python/director/drakevisualizer.py

```diff
@@ -215,62 +235,57 @@ def loadPolyDataMeshes(geom):
             print 'warning, cannot find file:', filename
             return []
 
+        meshDataList = []
+
         if filename.endswith('vtm'):
             polyDataList = ioUtils.readMultiBlock(filename)
+            meshDataList = Geometry.DeriveMeshVisList(geom, filename, polyDataList)
         else:
-            polyDataList = [ioUtils.readPolyData(filename)]
-
-        if USE_TEXTURE_MESHES:
-            for polyData in polyDataList:
-                Geometry.loadTextureForMesh(polyData, filename)
-
-
-        Geometry.MeshCache[filename] = list(polyDataList)
-
-        return polyDataList
+            if filename.endswith('obj'):
```

Comment:
I confirm that this is a bug. It can be replicated by running the `controlled_iiwa_world_demo`, which uses `/manipulation/models/iiwa_description/urdf/iiwa14_polytope_collision.urdf`. This model references .obj files that do not have associated .mtl files and should be colored based on what the URDF specifies. I will fix this ASAP.